automates/program_analysis/CAST2GrFN/visitors/cast_to_agraph_visitor.py

```diff
@@ -0,0 +1,449 @@
+        args = []
+        body = []
+        if(node.func_args != []):
+            args = self.visit_list(node.func_args)
+        if(node.body != []):
+            body = self.visit_list(node.body) 
+
+        node_uid = uuid.uuid4()
+        args_node = uuid.uuid4()
+        body_node = uuid.uuid4() 
+
+        self.G.add_node(node_uid,label="Function: "+node.name)
+        self.G.add_node(args_node, label="Arguments")
+        self.G.add_node(body_node, label="Body")
+
+        self.G.add_edge(node_uid,body_node)        
+        self.G.add_edge(node_uid,args_node)        
+        for n in args:
+            self.G.add_edge(args_node,n)
+        
+        for n in body:
+            self.G.add_edge(body_node,n)
+
+
+        return node_uid
+
+    @visit.register
+    def _(self, node: List):
+        """Visits List nodes. We visit all the elements and add them to
+        this node. This node's UID is returned."""
+        values = []
+        if(node.values != []):
+            values = self.visit_list(node.values)
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid,label="List")
+        for n in values:
+            self.G.add_edge(node_uid,n)
+
+        return node_uid
+
+    @visit.register 
+    def _(self, node: Loop):
+        """Visits Loop nodes. We visit the conditional expression and the
+        body of the loop, and connect them to this node in the graph.
+        This node's UID is returned."""
+        expr = self.visit(node.expr)
+        body = []
+        if(node.body != []):
+            body = self.visit_list(node.body)
+        node_uid = uuid.uuid4()
+        test_uid = uuid.uuid4()
+        body_uid = uuid.uuid4()
+
+        self.G.add_node(node_uid,label="Loop")
+        self.G.add_node(test_uid,label="Test")
+        self.G.add_node(body_uid,label="Body")
+
+        self.G.add_edge(node_uid,test_uid)
+        self.G.add_edge(test_uid,expr)
+        self.G.add_edge(node_uid,body_uid)
+        for n in body:
+            self.G.add_edge(body_uid,n)
+
+        return node_uid
+    
+    @visit.register
+    def _(self, node: ModelBreak):
+        """Visits a ModelBreak (break statment) node. 
+        The node's UID is returned"""
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid, label="Break")
+        return node_uid
+
+    @visit.register
+    def _(self, node: ModelContinue):
+        """Visits a ModelContinue (continue statment) node. 
+        The node's UID is returned"""
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid, label="Continue")
+        return node_uid
+
+    @visit.register
+    def _(self, node: ModelIf):
+        """Visits a ModelIf (If statement) node.
+        We visit the condition, and then the body and orelse 
+        attributes if we have any. They're all added to the Graph 
+        accordingly. The node's UID is returned."""
+        expr = self.visit(node.expr)
+        body = []
+        orelse = []
+        if(node.body != []):
```

Comment:
Im curious, do you need these if checks or can you just do 

body = self.visit_list(node.body)

Reply:
I put them in there because I recall having issues in other places when the checks weren't there, so I left them in just to be cautious. I can investigate more later and see if I truly need them, but I'd like to leave them there for now.